advance_check_void/models/account_payment.py

```diff
@@ -0,0 +1,391 @@
+                self.check_number = 0
+            elif self._context.get("is_vendor"):
+                if (
+                    self.payment_method_id.payment_type == "outbound"
+                    and self.payment_method_code
+                    in ("check_printing", "ACH-Out", "manual")
+                ):
+                    self.is_visible_check = False
+            elif (
+                self.payment_method_id.payment_type == "outbound"
+                and self.payment_method_code in ("check_printing", "ACH-Out", "manual")
+            ):
+                self.is_visible_check = True
+                if not self.check_manual_sequencing:
+                    self.is_readonly_check = True
+
+    def print_checks(self):
+        payment_check_void_obj = self.env["payment.check.void"]
+        for rec in self:
+            if rec.payment_type == "outbound":
+                domain = [
+                    ("payment_id", "=", rec.id),
+                    ("check_number", "=", int(rec.check_number)),
+                ]
+                payment_check_void_ids = payment_check_void_obj.search(
+                    domain, order="id DESC", limit=1
+                )
+                is_manual = rec.journal_id.check_manual_sequencing
+                if payment_check_void_ids and is_manual:
+                    rec.check_number = rec.journal_id.check_next_number
+                    rec.is_visible_check = True
+                    sequence = rec.journal_id.check_sequence_id
+                    sequence.next_by_id()
+                    message = (
+                        """<ul class="o_mail_thread_message_tracking">
+                        <li>(_(Check Updated Date:)) %s</li>
+                        <li>(_(Check Number:)) %s (Generated)</li>
+                        <li>(_(State:)) %s</li></ul>"""
+                    ) % (
+                        datetime.now().strftime(DEFAULT_SERVER_DATE_FORMAT),
+                        rec.check_number,
+                        rec.state.title(),
+                    )
+                    rec.message_post(body=message)
+        result = super(AccountPayment, self).print_checks()
+        return result
+
+    def void_check_button(self):
+        return {
+            "name": _("Void Check"),
+            "type": "ir.actions.act_window",
+            "res_model": "simple.void.check",
+            "view_type": "form",
+            "view_mode": "form",
+            "target": "new",
+            "context": {
+                "default_payment_id": self.id,
+            },
+        }
+
+    def write(self, vals):
+        if vals.get("check_number") and not str(vals.get("check_number")).isdigit():
+            raise ValidationError(_("Check number must be integer."))
+        result = super(AccountPayment, self).write(vals)
+        check_hist_obj = self.env["payment.check.history"]
+        for res in self:
+            new_chk_id = False
+            self._cr.execute(
```

Comment:
Can we replace this SQL with an ORM `write()`?
```suggestion
            self._cr.execute(
```